node/src/components/block_synchronizer.rs

```diff
@@ -898,6 +911,55 @@ impl BlockSynchronizer {
         }
     }
 
+    fn sync_leap_fetched(&mut self, result: Result<FetchedData<SyncLeap>, FetcherError<SyncLeap>>) {
+        let (block_hash, maybe_sync_leap, maybe_peer_id): (
+            BlockHash,
+            Option<Box<SyncLeap>>,
+            Option<NodeId>,
+        ) = match result {
+            Ok(FetchedData::FromPeer { item, peer }) => {
+                debug!(
+                    "BlockSynchronizer: fetched sync leap {:?} from peer {}",
+                    item.fetch_id().block_hash(),
+                    peer
+                );
+
+                (item.fetch_id().block_hash(), Some(item), Some(peer))
+            }
+            Ok(FetchedData::FromStorage { item }) => {
+                (item.fetch_id().block_hash(), Some(item), None)
+            }
```

Comment:
Sync leaps should never be fetched from storage, right? So this should be an error.

Reply:
addressed